pdm/openapi/action.yml

```diff
@@ -0,0 +1,118 @@
+name: Generate OpenAPI
+description: Generate and compare OpenAPI specifications
+
+inputs:
+  python-version:
+    description: Python version used to build
+    required: false
+    default: "3.11"
+  github-token:
+    description: A Github token with proper permissions
+    required: false
+    default: ${{ github.token }}
+  pypi-token:
+    description: Private PyPI token (JFrog read)
+    required: false
+  init:
+    description: Clone & sync
+    default: 'true'
+  group:
+    description: Dependency group(s) to install
+    default: docs
+  exclude-group:
+    description: Dependency group(s) to exclude from install
+    default: ''
+  spectral-file-glob:
+    description: Glob for Spectral to run on
+    default: 'docs/openapi.yaml'
+  spectral-ruleset:
+    description: Ruleset to enforce
+    default: '.github/openapi.spectral.yaml'
+  fail-on-lint:
+    description: Whether Spectral should fail on errors (`true`) or only warn (`false`)
+    default: 'true'
+
+runs:
+  using: composite
+  steps:
+  - name: Clone and install dependencies
+    uses: LedgerHQ/actions/pdm/init@main
+    id: init
+    if: inputs.init == 'true'
+    with:
+      group: ${{ inputs.group }}
+      exclude-group: ${{ inputs.exclude-group }}
+      pypi-token: ${{ inputs.pypi-token }}
+      python-version: ${{ inputs.python-version }}
+
+  - name: Generate the OpenAPI specifications
+    run: |
+      : Generate the OpenAPI specifications
+      pdm doc:openapi
+    env:
+      FORCE_COLOR: 'true'
+    shell: bash
+
+  - name: Run Spectral lint
+    uses: stoplightio/spectral-action@latest
+    with:
+      file_glob: ${{ inputs.spectral-file-glob }}
+      spectral_ruleset: ${{ inputs.spectral-ruleset }}
+      args: ${{ inputs.fail-on-lint == 'true' && '--fail' }}
+
+  - name: Upload the OpenAPI specifications
+    uses: actions/upload-artifact@v4
+    with:
+      name: openapi
+      path: docs/openapi.yaml
+
```

Comment:
Add a comment or a placeholder there to warn about the fact the specification are being moved and branch changed and so any additional step/test against the specs should be done before (so the next one trying to  work on this doesn't struggle on this)